### PR TITLE
embree: build fixes, mostly for macOS/arm64

### PR DESCRIPTION
**Pull request by aumuell**

### CODE REVIEW

File: var/spack/repos/builtin/packages/embree/package.py [hunk 1]
```diff
@@ -40,24 +42,33 @@ def cmake_args(self):
         args = [
             '-DBUILD_TESTING=OFF',
             '-DEMBREE_TUTORIALS=OFF',
+            '-DEMBREE_TBB_ROOT={0}'.format(spec['tbb'].prefix),
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/embree/package.py:45]:
```suggestion
            self.define('DEMBREE_TBB_ROOT', spec['tbb'].prefix),
```

File: var/spack/repos/builtin/packages/embree/package.py [hunk 2]
```diff
@@ -40,24 +42,33 @@ def cmake_args(self):
...
+                self.define('EMBREE_ISA_AVX2', 'avx512' not in spec.target),
+                self.define('EMBREE_ISA_AVX512{0}'.format(avx512_suffix), True),
+            ])
+
+            if spec.satisfies('%gcc@:7'):
+                # remove unsupported -mprefer-vector-width=256, otherwise copied
+                # from common/cmake/gnu.cmake
+                args.append('-DFLAGS_AVX512{0}=-mavx512f -mavx512dq -mavx512cd'
+                            ' -mavx512bw -mavx512vl -mf16c -mavx2 -mfma -mlzcnt'
+                            ' -mbmi -mbmi2'.format(avx512_suffix))
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/embree/package.py:72]:
```suggestion
                flags = ["avx512f", "avx512dq", "avx512cd", "avx512bw",
                         "avx512vl", "f16c", "avx2", "fma", "lzcnt", "bmi",
                         "bmi2"]
                args.append(self.define('FLAGS_AVX512' + avx512_suffix,
                                        ' '.join('-m' + f for f in flags))))
```

File: var/spack/repos/builtin/packages/embree/package.py [hunk 3]
```diff
@@ -40,24 +42,33 @@ def cmake_args(self):
...
+                avx512_suffix = 'SKX'
+            args.extend([
+                # code selection and defines controlling namespace names are based on
+                # defines controlled by compiler flags, so disable ISAs below compiler
+                # flags chosen by spack
+                self.define('EMBREE_ISA_SSE2', 'sse4_2' not in spec.target),
+                self.define('EMBREE_ISA_SSE42', 'avx' not in spec.target),
+                self.define('EMBREE_ISA_AVX', 'avx2' not in spec.target),
+                self.define('EMBREE_ISA_AVX2', 'avx512' not in spec.target),
+                self.define('EMBREE_ISA_AVX512{0}'.format(avx512_suffix), True),
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/embree/package.py:64]:
```suggestion
                self.define('EMBREE_ISA_AVX512' + avx512_suffix, True),
```

**Review by sethrj — APPROVED** [review]
Looks fine, but I have a few minor suggestions

**Comment by alalazo** [comment]
What is the state of this PR?

**Comment by aumuell** [comment]
this PR is made obsolete by #38083

**Pull request closed without merging**